python_modules/dagster/dagster/_daemon/backfill.py

```diff
@@ -10,6 +10,32 @@
 from dagster._utils.error import SerializableErrorInfo
 
 
+class BackfillLogAdapter(logging.LoggerAdapter):
+    """Logger that will always prepend the backfill id to the log message. Additionally it
+    adds the backfill_id to the `extra` dictionary of the log record so that it is included
+    as a key-value pair in JSON log formatting.
+
+    Usage:
+
+    :code-block: python
+
+        adapted_backfill_logger = BackfillLogAdapter(logger, {"backfill_id": backfill_id})
+    """
+
+    def process(self, msg, kwargs) -> Any:
+        backfill_id = self.extra["backfill_id"]
+
+        # the extra passed in the log call logger.info("message", extra={"foo": "bar"}), not
+        # to be confused with the extra passed to the BackfillLogAdapter constructor
+        # add the backfill_id to the extra too so that JSON log formatting includes it as
+        # a key-value pait
+        log_call_extra = kwargs.get("extra", {})
+        log_call_extra["backfill_id"] = backfill_id
+        kwargs["extra"] = log_call_extra
+
+        return f"Backfill {backfill_id}: {msg}", kwargs
```

Comment:
are json logs what gets sent to datadog? that's the most important spot to have the backfill id present.

what's your thinking on the reason to remove it from the default logs?

Reply:
Its a little easier to deal with json logs that have fixed messages and vary properties only as kv pairs. That said I think most modern systems apply some cute pattern matching to realize parts of messages that are ids so maybe a moot point. 

JSON logging is what we (and i assume most mature open source deployments) use

Reply:
ok that makes sense. I'm not strongly tied to having the id in the default logs, especially if we think most OSS deployments will have the logs in json anyway

Reply:
well to be clear, not most by volume but the ones that somebody has decided to invest in monitoring for

Reply:
ok - i still think we need to do this Adapter pattern so that we automatically add the extra to the logs, even if we remove the prepended message. just want to make sure that adding the Adapter seems ok?

Reply:
i think so, not an expert on logging by any means. Not sure if anything else will start getting picked up by [1] beyond our intention

Reply:
did a search through the code base. I see us using `extra` here https://github.com/dagster-io/dagster/blob/52c78304feb121edbc28e65f9974ade5af124a53/python_modules/dagster/dagster/_core/log_manager.py#L407-L410

here (which looks to just be passing extras through) https://github.com/dagster-io/dagster/blob/52c78304feb121edbc28e65f9974ade5af124a53/python_modules/dagster/dagster/_core/log_manager.py#L285

Reply:
doesn't raise any bells to me to include this stuff, but i also don't know logging that well. 

@smackesey it looks like you added the 
```
extra={ 
     LOG_RECORD_EVENT_ATTR: dagster_event, 
     LOG_RECORD_EVENT_BATCH_METADATA_ATTR: batch_metadata, 
 }, 
```
 lines, do you see any issue with these getting printed out when JSON logs are being used?

Reply:
those are for run logs, which I don't think any flow with `get_structlog_shared_processors` effects but im not certain

Reply:
ahh ok. i ran with `log-format json` and confirmed that those don't come through

Reply:
let's go with just adding the `extra` in this PR. if there later seems value in having the backfill id prepended to each message it will be easy to add back